dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/ndk/DatadogNdkCrashHandler.kt

```diff
@@ -0,0 +1,231 @@
+import com.datadog.android.rum.model.ViewEvent
+import com.google.gson.JsonSyntaxException
+import java.io.File
+import java.io.IOException
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.TimeUnit
+
+internal class DatadogNdkCrashHandler(
+    private val ndkCrashDataDirectory: File,
+    private val dataPersistenceExecutorService: ExecutorService,
+    private val asyncLogWriter: Writer<Log>,
+    private val asyncRumWriter: Writer<RumEvent>,
+    private val logGenerator: LogGenerator,
+    private val rumEventDeserializer: Deserializer<RumEvent> = RumEventDeserializer()
+) : NdkCrashHandler {
+
+    override fun handleNdkCrash() {
+        dataPersistenceExecutorService.submit {
+            checkAndHandleNdkCrashReport()
+        }
+    }
+
+    private fun checkAndHandleNdkCrashReport() {
+        if (ndkCrashDataDirectory.exists()) {
+            // check if there is an NDK report
+            val crashLogFiles =
+                ndkCrashDataDirectory.listFiles { _, name -> name == CRASH_LOG_FILE_NAME }
+            if (crashLogFiles != null && crashLogFiles.isNotEmpty()) {
+                readFromFile(
+                    crashLogFiles[0],
+                    NdkCrashLog::class.java,
+                    "Malformed ndk crash error log",
+                    "Error while trying to read the ndk crash log"
+                )?.let {
+                    handleNdkCrashLog(it, searchLastRumViewEvent())
+                }
+            }
+            clearCrashLog()
+        }
+    }
+
+    private fun searchLastRumViewEvent(): RumEvent? {
+        val lastViewEventFiles =
+            ndkCrashDataDirectory.listFiles { _, name ->
+                name == RumFileWriter.LAST_VIEW_EVENT_FILE_NAME
+            }
+        if (lastViewEventFiles != null && lastViewEventFiles.isNotEmpty()) {
+            return readFromFile(
+                lastViewEventFiles[0],
+                RumEvent::class.java,
+                "Malformed RUM ViewEvent log",
+                "Error while trying to read the last rum view event log"
+            )
+        }
+        return null
+    }
+
+    private fun <T> readFromFile(
+        file: File,
+        type: Class<T>,
+        parsingErrorMessage: String,
+        ioErrorMessage: String
+    ): T? {
+        return try {
+            val serializedData = file.readText(Charsets.UTF_8)
+            @Suppress("UNCHECKED_CAST")
+            if (type == RumEvent::class.java) {
+                rumEventDeserializer.deserialize(serializedData) as? T
+            } else {
+                fromJson(serializedData) as T
+            }
+        } catch (e: JsonSyntaxException) {
+            sdkLogger.e(parsingErrorMessage, e)
+            null
+        } catch (e: IOException) {
+            sdkLogger.e(ioErrorMessage, e)
+            null
+        }
+    }
+
+    private fun handleNdkCrashLog(ndkCrashLog: NdkCrashLog, lastRumViewEvent: RumEvent?) {
```

Comment:
This method is a bit long, can you split it in smaller chunks ?

Reply:
will try to see if I can split it more, I already created some separated methods from it.